### PR TITLE
Allow setting of 'iconFormat' in icon json so we can improve performance

### LINKED ISSUE #24: Add "iconFormat" setting to allow performance improvement
We're using a Pro icon set kit with Font Awesome and were able to create a series of JSON files. That does work, but the problem is that the resulting JSON files end up being over 6 MB in total because the files include thousands of SVGs. This makes things very slow when these files load.

In our case (and probably that of many others), we don't actually need the SVGs inline because we're including the Font Awesome CSS/JavaScript on the pages of our site where the icon picker is used. If the picker just used the `<i class='far fa-abacus'></i>` markup, it would work fine. Removing the SVGs from the JSON files would allow us to greatly reduce the file sizes and really improve performance.

### TL;DR

We're proposing a change to the JavaScript that will allow for a huge performance improvement related to downloading JSON files (95% smaller files), but will allow things to work just as they are now with existing Iconify files.

### Details

We're proposing adding an **_optional_** `iconFormat` setting to the JSON files that could have values like:

- `svg` (the default) - Nothing changes. `body` continues to have the full SVG.

Example JSON (snipped, unchanged from current format)
```json
{
	"prefix": "far fa-",
	"info": {
		"name": "Font Awesome Regular",
	},
	"lastModified": 1689174287,
	"icons": {
		"abacus": {
			"body": "<path d=\"M552 0c-13.25 ...  fill=\"currentColor\"/>",
			"width": 576
		},
```

- `i` - `body` is not needed at all. The picker would use markup like `<i class='far fa-abacus'></i>`.

Example JSON (snipped, `iconFormat` can be set to "i", `body` can be empty or missing entirely)
```json
{
	"prefix": "far fa-",
    "iconFormat": "i",
	"info": {
		"name": "Font Awesome Regular",
	},
	"lastModified": 1689174287,
	"icons": {
		"abacus": { 
            "body": "",
			"width": 576
		},
		"acorn": {
			"width": 448
		},
```

- `markup` - The picker would use the actual markup set in the `body`. This would allow the use of different, custom markup for icons, e.g. `<span class='far fa-abacus'></span>`.

Example JSON (snipped, `iconFormat` must be set to "markup", `body` must be set)
```json
{
	"prefix": "far fa-",
    "iconFormat": "markup",
	"info": {
		"name": "Font Awesome Regular",
	},
	"lastModified": 1689174287,
	"icons": {
		"abacus": {
            "body": "<i class='far fa-abacus'></i>",
			"width": 576
		},
```

**Comparison of JSON file sizes**

| iconFormat | Total file size | % of current | Notes |
| --- | --- | --- | --- | 
| svg | 6.1 M | -- | SVGs included (current) |
| markup | 616K | 10% | markup included in JSON |
| i | 364K | 6% | Using `"body": ""` |'
| i | 284K | 4.6% |  `body` not present in JSON |

By adding this **_optional_** field, we can reduce the size of the JSON files by as much as 95%. We've got a PR that does this, and we've been using it. For us, this changed it from "painfully slow" to "snappy".

The main part of the change to `src/js/IconPicker.js`:

```javascript
let iconElement;
if (iconFormat === 'i' || !value.body) {
    iconElement = document.createElement('i');
    iconElement.setAttribute('class', iconTarget.dataset.value);
}
else if (iconFormat === 'markup') {
    let t = document.createElement('template');
    t.innerHTML = value.body;
    iconElement = t.content;
}
else {
    iconElement = document.createElementNS('https://www.w3.org/2000/svg', 'svg');
    iconElement.setAttribute('height', '24');
    iconElement.setAttribute('width', '24');
    iconElement.setAttribute('viewBox', `0 0 ${value.width ? value.width : library.width} ${value.height ? value.height : library.height}`);
    iconElement.innerHTML = value.body;
}
```

The logic is such that:
1. If `iconFormat` is set to `i` or the `body` has no value, it will create markup like `<i class='far fa-abacus'></i>`, else...
2. If  `iconFormat` is set to `markup`, it will use whatever's in `body`, else...
3. If none of the above are done, it will do exactly what it did before-- use the SVG.

With this, **no changes need to be made to the existing Iconify files**. Those will still include the SVGs in the body, so they will continue to work as they have. But, if you wanted to download those files locally, you could run something like sed or just a search-and-replace on them to remove the entire "body" value from the JSON files and reduce the required downloads by 95%.

PR to follow.

### PR DESCRIPTION
**Pull request by jaybeaton**
See #24 

> We're proposing a change to the JavaScript that will allow for a huge performance improvement related to downloading JSON files (95% smaller files), but will allow things to work just as they are now with existing Iconify files.

### CODE REVIEW

**Review by Fredxd — CHANGES_REQUESTED** [review]
Hi,
Thanks for your PR.
Before merging it, can you do some changes please:

- Can you update the Readme (and demo if necessary) with the new features.
- Can you rebase from main before rebuild `dist` directory

Thanks

**Comment by jaybeaton** [comment]
@Fredxd 

I can rebase, etc., but the picker no longer works for me (even without my new changes) after merging the change in #13:

https://github.com/AppoloDev/vanilla-icon-picker/pull/13#issuecomment-1642327709

I think that change should be rolled back.

**Comment by simonmnt** [comment]
Hi @jaybeaton ,

I updated `develop`branch with this #13 problems.
Before merge, please update documentation with this new feature. (iconFormat parameter).

Regards

**Comment by jaybeaton** [comment]
@simonmnt @Fredxd 

I rebased from develop and rebuilt the `dist` directory. I also updated the README to document the new `iconFormat` setting.

**Review by simonmnt — APPROVED** [review]
